File: frontend/lib/types/test-case.ts
export interface TestCase {
  id: string
  requirementId: string
  name: string
  description?: string
  testData?: Record<string, any>
  expectedOutcome?: string
  status: 'pending' | 'passed' | 'failed'
  createdAt: string
  updatedAt: string
}

export interface CreateTestCaseRequest {
  name: string
  description?: string
  testData?: Record<string, any>
  expectedOutcome?: string
}

export interface UpdateTestCaseRequest {
  name?: string
  description?: string
  testData?: Record<string, any>
  expectedOutcome?: string
  status?: 'pending' | 'passed' | 'failed'
}

export const TEST_STATUSES = [
  { value: 'pending', label: 'Pending', color: 'bg-gray-100 text-gray-700' },
  { value: 'passed', label: 'Passed', color: 'bg-green-100 text-green-700' },
  { value: 'failed', label: 'Failed', color: 'bg-red-100 text-red-700' },
] as const

File: api/models/test_case.py
"""Test case model"""
from sqlalchemy import Column, String, DateTime, ForeignKey, JSON
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from datetime import datetime
import uuid

from api.config.database import Base


class TestCase(Base):
    __tablename__ = "test_cases"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    requirement_id = Column(UUID(as_uuid=True), ForeignKey("requirements.id", ondelete="CASCADE"), nullable=False)
    name = Column(String(255), nullable=False)
    description = Column(String)
    test_data = Column(JSON)
    expected_outcome = Column(String)
    status = Column(String(50), default="pending", nullable=False)
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)

    # Relationships
    requirement = relationship("Requirement", backref="test_cases")
